### ISSUE TITLE
Add built-in methods for comparing labels between alignment, tree and meta data

### ISSUE THREAD

**Issue by Wytamma** [phytest-devs/phytest#27]
We need a built in method for comparing labels between all the data structures. Here is a manual example from @sebastianduchene. 

```python
def test_aln_tree_match_names(alignment: Alignment, tree : Tree):
    tree_names = [i.name for i in tree.get_terminals()]
    aln_names = [i.name for i in alignment]
    have_same_number_of_taxa = len(tree_names) == len(aln_names)
    all_aln_names_in_tree = all([i in aln_names for i in tree_names])
    assert have_same_number_of_taxa and all_aln_names_in_tree
```

We also have this example in phytest-continuous-testing-example:

```python
def test_sequences_in_metadata(sequence: Sequence, data: DataFrame):
    # ensure all sequences are in the metadata
    data.assert_contains('ID', sequence.id)

def test_sequences_in_tree(sequence: Sequence, tree: Tree):
    # ensure all sequences are in the tree
    assert sequence.id in [clade.name.split('_')[0] for clade in tree.get_terminals()]
```

However, this method doesn't ensure the same number of labels and tips.

**Comment by Wytamma** [comment]
Could use set operations to do this e.g. 

```python
def test_aln_tree_match_names(alignment: Alignment, tree : Tree):
    tree_names = {i.name for i in tree.get_terminals()}
    aln_names = {i.name for i in alignment}
    assert len(tree_names.difference(aln_names)) == 0
```

**Comment by Wytamma** [comment]
Solved in dd611b1edf4203c2f12c615948d4f6e9de7b1714
